code/modules/admin/verbs/adminhelp.dm

```diff
@@ -78,24 +78,46 @@
 		return
 	if(src.handle_spam_prevention(msg,MUTE_ADMINHELP))
 		return
+	var/ref_mob = "\ref[mob]"
+	for(var/datum/adminticket/T in admintickets)
+		if(T.permckey == src.ckey && T.resolved == "No")
+			if(alert(usr,"You already have an adminhelp open, would you like to bump it?", "Bump Adminhelp", "Yes", "No") == "Yes")
+				T.logs += "[src.ckey] has bumped this adminhelp!"
+				if(T.admin == "N/A")
+					usr << "<b>Due to the fact your Adminhelp had no assigned admin, admins have been pinged.</b>"
+					message_admins("[src.ckey] has bumped their adminhelp #[T.ID], still no assigned admin!")
+					msg = "<span class='adminnotice'><b><font color=red>ADMINHELP: </font><A HREF='?priv_msg=[ckey];ahelp_reply=1'>[key_name(src)]</A> (<A HREF='?_src_=holder;adminmoreinfo=[ref_mob]'>?</A>) (<A HREF='?_src_=holder;adminplayeropts=[ref_mob]'>PP</A>) (<A HREF='?_src_=vars;Vars=[ref_mob]'>VV</A>) (<A HREF='?_src_=holder;subtlemessage=[ref_mob]'>SM</A>) (<A HREF='?_src_=holder;traitor=[ref_mob]'>TP</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=[ref_mob]'>FLW</A>):</b> [T.msg]</span>"
+					for(var/client/X in admins)
+						if(X.prefs.toggles & SOUND_ADMINHELP)
+							X << 'sound/effects/adminhelp.ogg'
+						X << msg
+				else
+					usr << "<b>Admins have been notified.</b>"
+					message_admins("[src.ckey] has bumped their adminhelp #[T.ID].")
+				src.verbs -= /client/verb/adminhelp
+				adminhelptimerid = addtimer(src,"giveadminhelpverb",1200, FALSE)
+				return
+			usr << "<b>Thank you for your patience.</b>"
+			return
+
+	src.verbs -= /client/verb/adminhelp
+	adminhelptimerid = addtimer(src,"giveadminhelpverb",1200, FALSE)
```

Comment:
i really dislike how it gives the delay even when you open the ahelp window and then click cancel,could you actually remove ahelp from src verbs and give it back only if an actual ahelp gets sent please